lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -589,11 +589,24 @@ def _run_script(self, rerun_data: RerunData) -> None:
                         exec(code, module.__dict__)
 
                     self._session_state.maybe_check_serializable()
+                    # setting the session state here triggers a yield-callback call
+                    # which reads self._requests and checks for rerun data
```

Comment:
I added this comment because it took me a while to understand the app flow in the unit-test and from where exactly the RerunException was thrown / the yield_callback called.